frontend/src/app/core/services/firebase-sign-in.service.ts

```diff
@@ -1,52 +1,158 @@
 import { Injectable } from '@angular/core';
-import { AngularFireAuth } from '@angular/fire/auth';
-import { UserService } from '@core/services/user.service';
-import { RegisterDialogService } from '@core/services/register-dialog.service';
 import { Router } from '@angular/router';
+import { RegisterDialogService } from '@core/services/register-dialog.service';
+import { UserService } from '@core/services/user.service';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { Providers } from '@shared/models/providers';
+import { LinkProvider } from '@shared/models/user/link-provider';
 import { auth } from 'firebase/app';
+import { RegistrationWarningComponent } from '../components/registration-warning/registration-warning.component';
 import { AuthenticationService } from './authentication.service';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class FirebaseSignInService {
 
+  private unsubscribe$ = new Subject<void>();
```

Comment:
What are you doing with this? It's never emited

Reply:
I was trying different ways, and forgot about it, but it doesnt influence on the result